### PR TITLE
upgrade to go 1.17

### PR DESCRIPTION
**Pull request by caleblloyd**
- upgrades to go 1.17

### CODE REVIEW

**Review by wallyqs — APPROVED** [review]

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: go.sum [hunk 1]
```diff
@@ -24,10 +24,14 @@ github.com/julienschmidt/httprouter v1.3.0 h1:U0609e9tgbseu3rBINet9P48AI/D3oJs4d
 github.com/julienschmidt/httprouter v1.3.0/go.mod h1:JR6WtHb+2LUe8TCKY3cZOxFyyO8IZAc4RVcycCCAKdM=
 github.com/klauspost/compress v1.13.4 h1:0zhec2I8zGnjWcKyLl6i3gPqKANCCn5e9xmviEEeX6s=
 github.com/klauspost/compress v1.13.4/go.mod h1:8dP1Hq4DHOhN9w426knH3Rhby4rFm6D8eO+e+Dq5Gzg=
+github.com/klauspost/compress v1.15.1 h1:y9FcTHGyrebwfP0ZZqFiaxTaiDnUrGkJkI+f583BL1A=
```
[the remark this answers is not in the document]
Comment (caleblloyd) [reply go.sum:17]: :+1: done

[the remark this answers is not in the document]
Comment (kozlovic) [reply go.sum:17]: For some reason, the push run now fails (I have recycled once and it fails at the same test)

[the remark this answers is not in the document]
Comment (caleblloyd) [reply go.sum:17]: yes, this test is not stable.  it passes some runs, fails some runs: https://github.com/nats-io/nats-account-server/runs/5605843310?check_suite_focus=true#step:6:292

[the remark this answers is not in the document]
Comment (kozlovic) [reply go.sum:17]: ah ok!

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**